src/list.rs

```diff
@@ -64,6 +64,28 @@ impl DirIter {
         }
         return Ok(Some(&*entry));
     }
+
+    /// Returns the current directory iterator position. The result should be handled as opaque value
+    pub fn current_position(&self) -> io::Result<libc::c_long> {
```

Comment:
I wouldn't expose `c_long`, I would wrap it into an opaque token. I don't think this number is meaningful by itself.

Reply:
Also, we should note this:
> Values returned by telldir()
>     are good only for the lifetime of the DIR pointer,	dirp, from which they
>     are derived.

This is from freebsd and macos manual.

Also I'm not sure maybe we can enforce this by:
```
fn current_position<'x>(&self) -> io::Result<Position<'x>> { ...}
```
Not sure if this is a great idea yet....